src/matrixLabels/gatherTextTriangles.js

```diff
@@ -59,16 +62,33 @@ export default function gather_text_triangles(store, viz_area, inst_axis) {
         if (labels.precalc[inst_axis]) {
           // calculate text vector
           // vectorize the label so we can draw it at any scale
-          inst_text_vect = vectorize_label(store, inst_axis, inst_name);
-          text_triangles[inst_axis][inst_name] = inst_text_vect;
-          // current and new offsets, based on reordering
-          inst_text_vect.inst_offset = [0, inst_label.offsets.inst];
-          inst_text_vect.new_offset = [0, inst_label.offsets.new];
-          text_triangles.draw[inst_axis].push(inst_text_vect);
+          tasks.push(vectorizeWorker.postMessage({
+            fontDetail,
+            name: inst_name,
+            axis: inst_axis,
+            offsetInst: inst_label.offsets.inst,
+            offsetNew: inst_label.offsets.new,
+          }));
+          store.dispatch(
+            dropFromLabelQueue({ queue: "low", axis: inst_axis, label: inst_name })
+          );
         }
       }
     }
   });
+  Promise.all(tasks).then((results) => {
```

Comment:
Not sure why it's not letting me put a comment on the whole block but we should wrap this `Promise.all` call in `if (tasks.length)`, and then the bare `store.dispatch` at the end of the file in an `else` statement, so we don't call both. It appears that that fixes the lack of label rendering -- however this error message appears once that change is made:
<img width="584" alt="image" src="https://user-images.githubusercontent.com/4188237/187973575-9bf2b6d8-341e-4686-84cf-909f9ea5187b.png">
I'll leave it to you to fix that error.

Reply:
Fix that error. But it still could not render automatically.
Just want to double check, once we sent the `mutateVisualizationState` action, the canvas should be updated right?